gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AllergyIntoleranceMapper.java

```diff
@@ -256,16 +278,90 @@ private void buildAllergyIntoleranceText(RCMRMT030101UK04ObservationStatement ob
                 allergyIntolerance.getCode().setText(originalTextFromCode);
             }
         }
+    }
+
+    private void buildNote(AllergyIntolerance allergyIntolerance,
+                           RCMRMT030101UK04ObservationStatement observationStatement,
+                           String compoundCode,
+                           CodeableConcept codeableConceptFromCode,
+                           CodeableConcept codeableConceptFromValue) {
+
+        var episodicityAnnotations = getEpisodicityAnnotations(observationStatement);
+        var pertinentInformationAnnotations = getPertinentInformationAnnotations(observationStatement);
+        var allergyAnnotation = getAllergyAnnotation(compoundCode, codeableConceptFromCode, codeableConceptFromValue);
+
+        var allergyIntoleranceNotes = allergyAnnotation == null
+                ? Stream.concat(episodicityAnnotations, pertinentInformationAnnotations)
+                : Stream.of(episodicityAnnotations, pertinentInformationAnnotations, allergyAnnotation)
+                    .flatMap(ain -> ain);
```

Comment:
Is this not just a different way of writing `Stream.concat`?

Reply:
If the `getAllergyAnnotation` returned an empty stream as opposed to null, then this could be simplified further too.

Reply:
stream.concat only supports two streams, so you would require 2 stream.concats to join 3 streams.  I'll try with including an empty stream and make sure it doesn't include a null entry

Reply:
Simplified this by returning empty stream.of for the allergyIntolleranceNotes, however still need use stream.of rather than stream.concat as there are 3 streams to merge